pipeline_steps/run_tests.groovy

```diff
@@ -0,0 +1,38 @@
+def run_irr_tests() {
+  common.conditionalStage(
+    stage_name: "Run IRR Tests",
+    stage: {
+      pubcloud.runonpubcloud {
+        try {
+          ansiColor('xterm') {
+            dir(env.WORKSPACE) {
+              print("Triggered by PR: ${env.ghprbPullLink}")
+              checkout([$class: 'GitSCM',
+                branches: [[name: "origin/pr/${env.ghprbPullId}/merge"]],
+                doGenerateSubmoduleConfigurations: false,
+                extensions: [[$class: 'CleanCheckout']],
+                submoduleCfg: [],
+                userRemoteConfigs: [
+                  [
+                    url: "https://github.com/${env.ghprbGhRepository}.git",
+                    refspec: '+refs/pull/*:refs/remotes/origin/pr/* +refs/heads/*:refs/remotes/origin/*'
+                  ]
+                ]
+              ]) // checkout
+              sh """#!/bin/bash
+              pushd \$(basename ${env.ghprbGhRepository})
```

Comment:
I don't think `checkout()` will create a subdir for the repo; `env.workspace` will become the checkout, so this pushd will probably fail. I think its generally a good idea to checkout to a subdir so that other repos can be checked out to their own subdirs if required in future.